packages/flutter/lib/src/widgets/image.dart

```diff
@@ -91,7 +117,11 @@ Future<void> precacheImage(
       if (!completer.isCompleted) {
         completer.complete();
       }
-      stream.removeListener(listener);
+      // Give callers at least one frame to subscribe to the image stream
+      // so that FlutterImageCache can weakly hold it without losing it.
+      SchedulerBinding.instance.scheduleFrameCallback((Duration timeStamp) {
```

Comment:
should probably be schedulePostFrameCallback so that you don't actually schedule a frame

Reply:
Done, and updated tests relevant to this to actually schedule a frame where needed.